### DISCUSSION TITLE
For not working in tasks?

### DISCUSSION THREAD

**Discussion by diegonayalazo** [babashka/babashka#1164] [Q&A]
BB Version 0.7.4
**platform**
Linux Fedora

**problem**

This is not working.

my-task {:doc "bug?" :task (for [j [ 1 2 3 ]] (prn j))}

Am I missing something? the for macro works ok in the command line but in tasks it doesn't seem to work.

**repro**

my-task {:doc "bug?" :task (for [j [ 1 2 3 ]] (prn j))}

**expected behavior**

This should print the vars in the vector.

Thanks Borkdude for Babashka :)

**Comment by borkdude** [comment]
This is not a bug, but you're dealing with laziness here.  Wrap `(doall ...)` around the for and you should see the output.
